plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -54,22 +57,18 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
 		loadedElements.clear();
 		loadingElements.clear();
 		errors.clear();
-		if (parentElement instanceof IWorkspaceRoot) {
-			return ConnectionsRegistrySingleton.getInstance().getAll();
-		} else if (parentElement instanceof ConnectionsRegistry) {
-			return ((ConnectionsRegistry) parentElement).getAll();
-//		} else if (parentElement instanceof ExpressConnection) {
-//			List<IDomain> domains = ((ExpressConnection) parentElement).getDomains();
-//			return domains.toArray(new IDomain[domains.size()]);
-//		} else if (parentElement instanceof com.openshift.kube.Client) {
-//			List<Project> openshiftProjects = ((com.openshift.kube.Client) parentElement).list(ResourceKind.Project);
-//			openshiftProjects.toArray(new Project[openshiftProjects.size()]);
-//		} else if (parentElement instanceof Project) {
-//			Project p = (Project) parentElement;
-//			List<Resource> resources = new ArrayList<Resource>(p.getResources(ResourceKind.DeploymentConfig));
-//			return resources.toArray();
+		if(parentElement instanceof ConnectionsRegistry){
+			ConnectionsRegistry registry = (ConnectionsRegistry) parentElement;
+			IConnection[] all = registry.getAll();
+			List<IConnection> connections = new ArrayList<IConnection>(all.length);
```

Comment:
see comment above, if getting all connections of a given type is in ConnectionsRegistry we can avoid duplicate the code here, a simple ConnectionsRegistry.getAll(Connection.class).toArray() is enough.